Dockerfile

```diff
@@ -1,19 +1,41 @@
-FROM node:14-alpine
+# application builder
+FROM node:16-alpine AS builder
 
-# directory for the app in the container
 WORKDIR /usr/app
 
-# copies all the app's files from host into the container folder which
-# might include the node_modules dir if npm install executed in the host
+# copy package.json and package-lock.json files
+COPY package*.json .
+
+# install dependencies including local development tools
+RUN npm install
+
+# copy the rest of the content
 COPY . /usr/app
 
-# removes any existing node_modules folder
-# this prevents the host's node_modules from being used in the container
-# which could cause issues with native binaries such as node_sass.
-RUN rm -rf /usr/app/node_modules/
+# build the application and generate a distribution package
+RUN npm run build
+
+# application package
+FROM node:16-alpine AS app
 
+WORKDIR /usr/app
+
+ENV NODE_ENV=production
 ENV CYPRESS_INSTALL_BINARY=0
 
-RUN npm install -g npm@7.21.0 && \
-    npm install && \
-    npm run i18n:get-translations
+# copy the package.json and package-lock.json files
+COPY package*.json .
+
+RUN npm install
```

Comment:
I think we can use [`npm ci`](https://blog.npmjs.org/post/171556855892/introducing-npm-ci-for-faster-more-reliable) for better reproducibility and reliability. Additionally we can pass `--only=prod` to avoid installing dev dependencies (as long as our dev dependencies are correctly deliniated, they may not be 😅 )

Reply:
Great suggestion @sarayourfriend and `npm ci --only=prod` isn't currently working, so I'll make an issue and PR to move the misplaced `devDependencies` into the `dependencies` block 👍

Reply:
Added in #401